course_catalog/models.py

```diff
@@ -74,12 +88,105 @@ class Course(TimestampedModel):
     url = models.URLField(null=True, max_length=2048)
     program_type = models.CharField(max_length=32, null=True, blank=True)
     program_name = models.CharField(max_length=256, null=True, blank=True)
-    location = models.CharField(max_length=128, null=True, blank=True)
     instructors = models.ManyToManyField(
         CourseInstructor, blank=True, related_name="courses"
     )
     learning_resource_type = models.CharField(
         max_length=20, default=ResourceType.course.value
     )
-    topics = models.ManyToManyField(CourseTopic, blank=True, related_name="courses")
-    prices = models.ManyToManyField(CoursePrice, blank=True, related_name="courses")
+    prices = models.ManyToManyField(CoursePrice, blank=True)
+
+
+class Bootcamp(LearningResource):
+    """
+    Bootcamp model for bootcamps. Being course-like, it shares a large overlap with Course model.
+    """
+
+    course_id = models.CharField(max_length=128, unique=True)
+    language = models.CharField(max_length=128, null=True, blank=True)
+    year = models.IntegerField(null=True, blank=True)
+    full_description = models.TextField(null=True, blank=True)
+    start_date = models.DateTimeField(null=True, blank=True)
+    end_date = models.DateTimeField(null=True, blank=True)
+    enrollment_start = models.DateTimeField(null=True, blank=True)
+    enrollment_end = models.DateTimeField(null=True, blank=True)
+    image_src = models.URLField(max_length=400, null=True, blank=True)
```

Comment:
This and other image-fields should be an `ImageField` so it can take advantage of our S3 image hosting and cloudfront distributions (look at `channels.models.Channel` for examples)

Reply:
Will do :+1:

Reply:
Question: should we maybe just use `ImageField` for user-created `LearningPaths` for now, instead of doing the migration of images/thumbnails for all Courses and Bootcamps to S3 right now as well? Right now we're getting URLs for hosted images directly from where we're importing Courses (e.g. edx, OCW). These image files are hosted elsewhere, but they may have CDN infrastructure behind them already. In any case, does it make sense to do this migration in another PR later?

Reply:
Do `program_image_upload_uri` and `learning_path_image_upload_uri` look about right to you?

Reply:
Oh, are these image urls provided in the upstream data as opposed to being a piece of content uploaded within discussions? If yes, ignore this piece of feedback.

Reply:
Although a comment on the field would probably be helpful

Reply:
For user-created `LearningPaths`, we discussed users can pick a image from a selection if they make their list public. We haven't yet discussed whether we want to let a user to upload a custom image for their `LearningPaths` . @pdpinch would you like to weigh in?

Reply:
Custom image would be great

Reply:
Then the change I made in LearningPath to use `ImageField` will allow that (@rhysyngsun, yes?)
And we want to keep `URLField` for the image urls we are getting from Course and Bootcamp import sources for now @Ferdi ?

Reply:
That sounds right to me - `ImageField` is for image assets that discussions handles and `URLField` is for externally managed assets we're referencing.